README.md

```diff
@@ -2,57 +2,91 @@
 
 ## Run
 
-To run the project it is recommended to do:
+### Setup
 
-- `poetry shell`
-- `poetry install`
-- `pip install -r dev-requirements.txt`
+It's recommended to use a virtual environment to run the project. To do so, follow these steps:
 
-This should allow to run `run_tests.sh`.
+``` bash
+poetry shell
+```
+
+Install the project dependencies:
+
+```bash
+
+    poetry install
+```
+
+Check if the project is working:
+
+```bash
+
+    jasm --help
+```
 
 ## Parse an assembly file
 
-`python3 main.py -p <pattern.yaml> -s <assembly_file.s>`
+```bash
+jasm -p <pattern.yaml> -s <assembly_file.s>
+```
 
 ## Parse a binary file
 
-`python3 main.py -p <pattern.yaml> -b <binary_file.bin>`
+```bash
+jasm -p <pattern.yaml> -b <binary_file.bin>
+```
 
-## Help
+## Use of macros
 
-To see the help run `python3 main.py -h` or `python3 main.py --help`
+You can also specify a macro file which will be used as the macros definitions. Examples of these files can be seen in `tests/macros/`.
 
-## Run tests
+```bash
+jasm -p <pattern.yaml> -s <assembly_file.s> --macros <macro_file.yaml>
+```
 
-To run test you can,
+The macro `tests/macros/jasm_macros.yaml` has the `@any` macro for example, which is used to match with any mnemonic or any operand.
 
-- Run `sh run_tests.sh` while inside `poetry shell`.
+### Macro Naming convention
 
-- Another options is to simply call `pytest`
+Macro names must start with the "@" identifier. Here is an example:
+
+```yaml
+macros:
+  - name: "@macro_1"
+    pattern: "replace_1"
+```
+
+## Help
+
+To see the help run `jasm -h` or `jasm --help`
+
+## Run tests
+
+To run test you just call `pytest`
 
 ## Manual Rules
 
 The rules for using in the patterns are as follows:
 
-- `$and`: Matches all the command in the list
+* `$and`: Matches all the command in the list
 
-- `$or`: Matches any of the command in the list
+* `$or`: Matches any of the command in the list
 
-- `$not`: Matches any command that is not in the list
+* `$not`: Matches any command that is not in the list
 
-- `$and_any_order`: Matches all the commands in the list in any order
+* `$and_any_order`: Matches all the commands in the list in any order
 
-- `@any`: Matches any command
+* `@any`: Matches any command
 
-- `$deref`: Used for dereferencing a register
+* `$deref`: Used for dereferencing a register
 
 NOTE: the `$deref` commands require the following syntax, using the example below:
 
-```
+```yaml
     - $deref:
         main_reg: "%rax"
         constant_offset: "0x0"
-        register_multiplier: "%rbx"
+        register_multiplier: "%rax"
```

Comment:
Can we leave `rbx` here so we have two diferents registers and thus, it is more clear which one is which?